devicetypes/smartthings/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,402 @@
+        
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"toggle", type:"generic",  decoration:"flat", width:6, height:4) {
+			tileAttribute ("device.lock", key:"PRIMARY_CONTROL") {
+				attributeState "locked", label:'locked', action:"lock.unlock", icon:"st.locks.lock.locked", backgroundColor:"#00A0DC", nextState:"unlocking"
+				attributeState "unlocked", label:'unlocked', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
+				attributeState "unknown", label:"unknown", icon:"st.locks.lock.unknown", backgroundColor:"#ffffff"
+                attributeState "unlocking", label:'unlocking', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
+                attributeState "autolockError", label:'autolock error', icon:"st.locks.lock.unlocked", backgroundColor:"#ff0000"
+			}
+            tileAttribute("device.displayName", key: "SECONDARY_CONTROL") {
+				attributeState "displayName", label: 'Model:  ${currentValue}'
+            }
+		}
+		valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:6, height:2) {
+			state "battery", label:'${currentValue}% BATTERY', unit:""
+		}
+		main "toggle"
+		details(["toggle", "battery"])
+	}
+}
+
+// Globals - Cluster IDs
+private getCLUSTER_POWER() { 0x0001 }
+private getCLUSTER_DOORLOCK() { 0x0101 }
+private getCLUSTER_ALARM() { 0x0009 }
+
+// Globals - Command IDs
+private getDOORLOCK_CMD_UNLOCK_DOOR() { 0x1F }
+private getDOORLOCK_RESPONSE_OPERATION_EVENT() { 0x20 }
+private getPOWER_ATTR_BATTERY_VOLTAGE() { 0x0020 }
+private getDOORLOCK_ATTR_LOCKSTATE() { 0x0000 }
+private getDOORLOCK_ATTR_DOORSTATE() { 0x0003 }
+private getALARM_ATTR_ALARM_COUNT() { 0x0000 }
+private getALARM_CMD_ALARM() { 0x00 }
+
+/**
+ * Called on app installed
+ */
+def installed() {
+	log.trace "ZigBee DTH - Executing installed() for device ${device.displayName}"
+}
+
+/**
+ * Called on app uninstalled
+ */
+def uninstalled() {
+	def deviceName = device.displayName
+	log.trace "ZigBee DTH - Executing uninstalled() for device $deviceName"
+	sendEvent(name: "lockRemoved", value: device.id, isStateChange: true, displayed: false)
+}
+
+
+/**
+ * Ping is used by Device-Watch in attempt to reach the device
+ */
+def ping() {
+	log.trace "ZigBee DTH - Executing ping() for device ${device.displayName}"
+    refresh()
+}
+
+/**
+ * Called when the user taps on the refresh button
+ */
+def refresh() {
+	log.trace "ZigBee DTH - Executing refresh() for device ${device.displayName}"
+	def cmds = zigbee.readAttribute(CLUSTER_DOORLOCK, DOORLOCK_ATTR_DOORSTATE)
+	log.info "ZigBee DTH - refresh() returning with cmds:- $cmds"
+	return cmds
+}
+
+
+/**
+ * Configures the device to settings needed by SmarthThings at device discovery time
+ *
+ */
+def configure() {
+	log.trace "ZigBee DTH - Executing configure() for device ${device.displayName}"
+	def cmds = zigbee.command(0x0000, 0x1E, "",[mfgCode: 0003]) // read modelName of the device as it is not being sent with zbjoin.
+    state.configured = true
```

Comment:
It doesn't look like this is used so it can be removed

Reply:
Okay, we will remove this line for the final delivery.